core/src/main/java/org/infinispan/configuration/global/SerializationConfigurationBuilder.java

```diff
@@ -136,7 +136,8 @@ public SerializationConfigurationBuilder classResolver(Object classResolver) {
    }
 
    public SerializationConfigurationBuilder addContextInitializer(SerializationContextInitializer sci) {
-      attributes.attribute(SERIALIZATION_CONTEXT_INITIALIZERS).computeIfAbsent(ArrayList::new).add(sci);
+      if (sci != null)
```

Comment:
Why do we need to hide this ? Can't we just let it NPE or throw an IllegalArgumentException ?

Reply:
I think I did this because it made the code in some tests simpler, but the tests shouldn't really influence the user code in this way. I'll update it to throw a CacheConfigurationException if sci is null.